pyccel/ast/builtins.py

```diff
@@ -774,6 +776,49 @@ def __str__(self):
         return "{args} -> {expr}".format(args=self.variables,
                 expr = self.expr)
 
+#==============================================================================
+class PythonType(Basic):
+    """ Represents the python builtin type function
+    """
+    __slots__ = ('_dtype','_precision','_obj')
+    _attribute_nodes = ('_obj',)
+
+    def __init__(self, obj):
+        self._dtype = obj.dtype
+        self._precision = obj.precision
+        self._obj = obj
+
+        if obj.rank > 0:
+            raise PyccelError("Python's type function doesn't return enough information about this object for pyccel to fully define a type")
+        super().__init__()
+
+    @property
+    def dtype(self):
+        """ Returns the dtype of this type
+        """
+        return self._dtype
+
+    @property
+    def precision(self):
+        """ Returns the precision of this type
+        """
+        return self._precision
+
+    @property
+    def arg(self):
+        """ Returns the object for which the type is determined
+        """
+        return self._obj
+
+    @property
+    def print_string(self):
+        """ Return a literal string representing the type that
+        can be used in a print  statement
+        """
+        return LiteralString("<class '{dtype}{precision}'>".format(
+            dtype = str(self.dtype),
+            precision = self.precision*8 if self.precision else ''))
```

Comment:
I don't know why we should print all this information instead of printing just the type same as python
```python
In [1]: type(1)
Out[1]: int

In [2]: a = 3.4

In [3]: type(a)
Out[3]: float

In [4]: import numpy as np

In [5]: type(np.array([2,3]))
Out[5]: numpy.ndarray

In [6]: def func():
   ...:     pass
   ...:

In [7]: type(func)
Out[7]: function

In [8]: type('Azul')
Out[8]: str

In [9]: t = type('c')

In [10]: type(t)
Out[10]: type
```

Reply:
You are confusing the result of `type(a)` and the result of `print(type(a))` which is equivalent to `str(type(a))`. See:
```python
In [1]: print(type(1))                                                                                                                                                                                     
<class 'int'>

In [2]: a = 3.4                                                                                                                                                                                            

In [3]: print(type(a))                                                                                                                                                                                     
<class 'float'>

In [4]: import numpy as np                                                                                                                                                                                 

In [5]: print(type(np.array([2,3])))                                                                                                                                                                       
<class 'numpy.ndarray'>

In [6]: def func(): 
   ...:     pass 
   ...:                                                                                                                                                                                                    

In [7]: print(type(func))                                                                                                                                                                                  
<class 'function'>

In [8]: print(type('Azul'))                                                                                                                                                                                
<class 'str'>

In [9]: t = type('c')                                                                                                                                                                                      

In [10]: print(type(t))                                                                                                                                                                                    
<class 'type'>
```

Reply:
This is related to the discussion with @MostafaMamouni above. It would be useful if you could give your opinion on this discussion

Reply:
I just read your disscusion about that, for me, I see that the first option is more convenient, because we are dealing with python users seeking performance, not all users are familiar with C or Fortran, so I think we need to keep things the same or nearby as python is.

Reply:
So then you are happy with the current output?

Reply:
yes.